tests/integration/cli_resource_negative_test.go

```diff
@@ -0,0 +1,222 @@
+			{
+				VerificationMethodId: signInputs[0].VerificationMethodId,
+				PrivKey:              testdata.GenerateByteEntropy(),
+			},
+		}, testdata.BASE_ACCOUNT_1)
+		Expect(err).To(HaveOccurred())
+
+		// Finally, create the resource
+		res, err := cli.CreateResource(collectionId, resourceId, resourceName, resourceType, resourceFile, signInputs, testdata.BASE_ACCOUNT_1)
+		Expect(err).To(BeNil())
+		Expect(res.Code).To(BeEquivalentTo(0))
+	})
+
+	It("cannot query a resource with missing cli arguments, non-existing collection, non-existing resource", func() {
+		collectionId2 := uuid.NewString()
+		resourceId2 := uuid.NewString()
+
+		// Fail to query a resource with missing cli arguments
+		//   a. missing collection id, resource id
+		_, err := cli.QueryResource("", "")
+		Expect(err).To(HaveOccurred())
+
+		//   b. missing collection id
+		_, err = cli.QueryResource("", resourceId2)
+		Expect(err).To(HaveOccurred())
+
+		//   c. missing resource id
+		_, err = cli.QueryResource(collectionId2, "")
+		Expect(err).To(HaveOccurred())
+
+		// Fail to query a resource with non-existing collection id
+		_, err = cli.QueryResource(collectionId2, resourceId)
+		Expect(err).To(HaveOccurred())
+
+		// Fail to query a resource with non-existing resource id
+		_, err = cli.QueryResource(collectionId, resourceId2)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("cannot query all resource versions with missing cli arguments, non-existing collection, non-existing resource", func() {
+		collectionId2 := uuid.NewString()
+		resourceName2 := rand.Str(10)
+
+		// Fail to query all resource versions with missing cli arguments
+		//   a. missing collection id, resource name
+		_, err := cli.QueryAllResourceVersions("", "")
+		Expect(err).To(HaveOccurred())
+
+		//   b. missing collection id
+		_, err = cli.QueryAllResourceVersions("", resourceName)
+		Expect(err).To(HaveOccurred())
+
+		//   c. missing resource name
+		_, err = cli.QueryAllResourceVersions(collectionId2, "")
+		Expect(err).To(HaveOccurred())
+
+		// Fail to query all resource versions with non-existing collection id
+		_, err = cli.QueryAllResourceVersions(collectionId2, resourceName)
+		Expect(err).To(HaveOccurred())
+
+		// Fail to query all resource versions with non-existing resource name
+		res, err := cli.QueryAllResourceVersions(collectionId, resourceName2)
+		Expect(err).To(BeNil())
+		Expect(len(res.Resources)).To(BeEquivalentTo(0))
+	})
+
+	It("cannot query resource collection with missing cli arguments, non-existing collection id", func() {
+		collectionId2 := uuid.NewString()
+
+		// Fail to query resource collection with missing cli arguments
+		//   a. missing collection id
+		_, err := cli.QueryResourceCollection("")
+		Expect(err).To(HaveOccurred())
+
+		// Fail to query resource collection with non-existing collection id
+		_, err = cli.QueryResourceCollection(collectionId2)
+		Expect(err).To(HaveOccurred())
+	})
+})
```

Comment:
Quite a few different scenarios but hard to understand which ones passed and didn't. (Unless I'm missing some output file somewhere?)

Reply:
There are relationships among different bundled cases.
That's why a procedural / functional style allows for less repetition.

One refactor that allows to block-scope procedural scenarios is to use `Context()` and enclose assertions in different `It()` leaf nodes.